[Rena] iP

Chiarena opened the pull request:
# **Duke**
>"The key is not to prioritize whats on your schedule, but to schedule your priorities." – Stephen Covey ([source](https://everydaypower.com/quotes-for-efficiency/))

Duke helps you record your tasks and makes it easier for you to manage them.
- Fast
* Easy
+ Fun

All you need to do is,
1. download it from [here](https://github.com/Chiarena/ip/releases/tag/A-Jar).
2. double click it.
3. add your tasks.
4. let it manage your tasks for you :wink:.

And it is **FREE**!

Features:

- [ ] Adding, deleting, marking tasks 
- [ ] Manage todo, deadlines and events
- [ ] View list of tasks and find tasks

If you are a Java programmer, you can use it to practice Java too. Here's tje main method:
~~~
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
~~~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Chiarena committed:
Level 0-1

Chiarena committed:
Level 1-2

Chiarena committed:
level 2 done

Chiarena committed:
Added level 3

Chiarena committed:
Added level 4 - deadline,todo,event

Chiarena committed:
added functions

Chiarena committed:
Automated testing done

Chiarena committed:
Exceptions class added

Chiarena committed:
Delete function added

Darkarche3 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+    protected String by;
```
You can consider making this private instead.

Darkarche3 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,160 @@
+import java.util.ArrayList;
+import java.util.Scanner;
 public class Duke {
```
Consider  changing name from Duke to the name of your bot!

Darkarche3 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,160 @@
+        return x + result + y;
+    }
+
+    public static String printMark(Task task) {
```
good that you followed camelCase.

Darkarche3 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+    public Boolean getStatus() { return isDone; }
+
+    public void markAsDone() {
+        this.isDone = true;
```
Good job remembering naming convention for boolean variables

Darkarche3 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,160 @@
+        }
+    }
+
+    public static String added(Task task, int count) {
```
Consider changing the name of this method to addTask

cocoanautz replied:
```suggestion
    public static String printAdded(Task task, int count) {
```

Methods should have verbs as names.

Darkarche3 reviewed (commented):
Well done for the most part. Just minor issues with your code.

NewtonKoh commented on src/main/java/Duke.java:
```diff
@@ -88,19 +101,22 @@ public static void main(String[] args) {
+                    } else if (message.startsWith("unmark")){
                         current.unmark();
                         System.out.println(printMark(current));
+                    }else{
```
I think you may have forgotten to add a space before and after the 'else' here.

NewtonKoh reviewed (commented):
LGTM. Small nits only

cocoanautz commented on src/main/java/Duke.java:
Could probably store the lines as a constant value to make the code easier to change.

cocoanautz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,160 @@
+                "____________________________________________________________\n";
+    }
+
+    public static ArrayList<Task> delete(ArrayList<Task> a, int count) {
```
The method name could be changed to explain more about what this method is deleting.

cocoanautz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,160 @@
+        Scanner input = new Scanner(System.in);
+        System.out.println("Enter Message");
+
+        final int ArraySize = 100;
```
Constants should be in screaming_snake_case.

cocoanautz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,160 @@
+                        String task = getTask(message);
+                        String[] parts = time.split("by");
+//                        tasks[counter] = new Deadline(task, parts[1]);
+                        tasks.add(new Deadline(task, parts[1]) );
```
```suggestion
                        tasks.add(new Deadline(task, parts[1]));
```

cocoanautz reviewed (commented):
Just some nits that require attention, mainly related to coding standards and naming.

Chiarena committed:
Added save feature

HusseinSafwan02 commented on src/main/java/DataManager.java:
```diff
@@ -0,0 +1,95 @@
+        String status = " ";
+        if(task instanceof Todo) {
+            symbol = "T";
+        }else if(task instanceof Deadline) {
```
could insert a space between } and else if

HusseinSafwan02 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,163 @@
+                    if (message.startsWith("todo")) {
+                        String[] parts = message.split(" ", 2);
+                        String task = parts[1];
+//                        tasks[counter] = new Todo(task);
```
consider removing these comments once code is finalised :)

HusseinSafwan02 reviewed (commented):
Some syntax issues to fix, as well as naming conventions, but well done so far!

Chiarena committed:
date time added

Chiarena committed:
oop done

Chiarena committed:
Classes arranged into packages

Chiarena committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Chiarena committed:
isExit corrected

Chiarena committed:
corrected bye command to terminate program

Chiarena committed:
JUnit tests added

Chiarena committed:
java docs added

Chiarena committed:
removed redundant code

Chiarena committed:
Find feature added

Chiarena committed:
Coding standard improved

Chiarena committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:conflict in Duke.java
#	src/main/java/Duke/Duke.java

Chiarena committed:
Merge branch 'branch-A-CodingStandard'

Chiarena committed:
Merge branch 'branch-Level-9'

# Conflicts: Conflicts between level-9 and master branches
#	src/main/java/Duke/Duke.java
#	src/main/java/actions/Parser.java
#	src/main/java/actions/TaskList.java

Chiarena committed:
further corrected code after merging

Chiarena committed:
Add critical assertions across to the parser class

Introduced assertions in the Parser class for handle methods to validate assumptions about method inputs.

This enhances debugging and code reliability during development.

Chiarena committed:
The code for each class is edited and sectioned to improve code quality

Arrow head code is reduced

These changes enhance maintainability and readability of the code

Chiarena committed:
Merge pull request #3 from Chiarena/branch-A-Assertions

Add critical assertions across to the parser class

Chiarena committed:
Merge branch 'master' into branch-CodeQuality

Chiarena committed:
Merge pull request #2 from Chiarena/branch-CodeQuality

CodeQuality

Chiarena committed:
update feature added

Chiarena committed:
Update README.md

Chiarena committed:
README edited

Chiarena committed:
Added user guide

Chiarena committed:
JavaFX

Chiarena committed:
Merge branch 'branch-Level-10'

Chiarena committed:
JavaFX edited
